lib/chef/provider/package/windows.rb

```diff
@@ -34,9 +34,16 @@ class Windows < Chef::Provider::Package
 
         require 'chef/provider/package/windows/registry_uninstall_entry.rb'
 
+        def define_resource_requirements
+          requirements.assert(:install) do |a|
+            a.assertion { new_resource.source unless package_provider == :msi }
+            a.failure_message Chef::Exceptions::NoWindowsPackageSource, "Source for package #{new_resource.name} must be specified in the resource's source property for package to be installed because the package_name property is used to test for package installation state for this package type."
```

Comment:
I guess you could add a "the" here:

"Source for package #{new_resource.name} must be specified in the resource's source property for package to be installed because the package_name property is used to test for **the** package installation state for this package type."